pkg/urlutil/git_test.go

```diff
@@ -9,6 +9,7 @@ var (
 		"git@bitbucket.org:atlassianlabs/atlassian-docker.git",
 		"https://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git",
+		"http://github.com/docker/docker.git#branch",
```

Comment:
what I meant was to make sure your logic parsed it correctly, meaning it maps to master + a dir - I wasn't concerned about the go-lang matches() method.  But I see you added a test for it down below, so we're ok.